lib/mongo/crypt/binary.rb

```diff
@@ -72,6 +72,28 @@ def close
 
         true
       end
+
+      # Convenient API for using binary object without having
+      # to perform cleanup.
+      #
+      # @example
+      #   Mongo::Crypt::Binary.with_binary([73, 76, 111, 118, 101, 82, 117, 98, 121]) do |binary|
+      #     binary.to_bytes # => [73, 76, 111, 118, 101, 82, 117, 98, 121]
+      #   end
+      #
+      # @since 2.12.0
+      def self.with_binary(data)
+        binary = self.new(data)
+
+        begin
+          yield(binary)
+        rescue => e
+          binary.close
+          raise e
+        end
```

Comment:
The pattern for this can be simpler:

<pre>
begin
  yield binary
ensure
  binary.close
end
</pre>